Checagem de SourceList e Correção de Bug

Dokaoista opened the pull request:
Neste PR, foram feitas as seguintes alterações:
- Adicionado a validação da existência do arquivo _infosectools.list_ na parte de instalação de pacotes;
- Corrigido o bug que se encontrava na parte de adicionar repositório, o qual se deve a somente a leitura sem checagem do arquivo _infosectools.list_.

Dokaoista committed:
Corrigido Bug FileNotFoundError

O erro se encontra na hora de adicionar repositório, no qual esta parte do código apenas lia o arquivo, e não checava a existência do mesmo.

Dokaoista committed:
Validação de SourceList

Feita a validação do arquivo infosectools.list na parte de instalção de pacotes.

stephan-lopes commented on instalacao.py:
```diff
@@ -13,6 +13,10 @@ def instalar_categoria(numero_da_categoria, instalador, argumento_do_instalador)
 def instalacao_de_pacotes(escolha, categorias, instalador, argumento_do_instalador):
 	try:
+
+		with open('/etc/apt/sources.list.d/infosectools.list', 'r') as arquivo:
```
Dá uma olhada nesse carinha aqui: https://www.freecodecamp.org/news/how-to-check-if-a-file-exists-in-python/

Acho que dessa forma, fica melhor a checagem de existência de arquivos do que tentar abrir ele

stephan-lopes commented on menu_subfunctions.py:
```diff
@@ -6,14 +6,15 @@ def add_repo_to_infosectools_list():
-    with open(file_path, "r") as file:
-        lines = file.readlines()
-    
-    if repository not in lines:
```
Na linha 13, ele escreve o repositorio dentro do source list caso não exista o arquivo, ou caso essa linha não exista dentro do arquivo. Sendo assim, remover ela vai fazer com que não seja adicionada as linhas do repositório do kali.

Dokaoista committed:
Update menu_subfunctions.py

Desta forma, o código checa a existência do arquivo e, logo depois, se o repositório está presente no mesmo.

Dokaoista committed:
Update instalacao.py

Alterado a forma de checagem para otimização e utilização da biblioteca "os".

Willianjesusdasilva commented on instalacao.py:
```diff
@@ -13,14 +13,23 @@ def instalar_categoria(numero_da_categoria, instalador, argumento_do_instalador)
-			print('Instalando pacotes...')
-			instalar_categoria(int(escolha), instalador, argumento_do_instalador)
+
+		if path.exists('/etc/apt/sources.list.d/infosectools.list'):
```
salve gatão.

No python, primeiro fazemos a tentativa e depois tratamos o erro.
Não é uma boa pratica em python fazer verificação antes de realizar uma ação dada a velocidade do intepretador.

Willianjesusdasilva reviewed (approved):
mesmo com a pendencia da verificação, é o PR está valido.
